assets/js/modules/analytics-4/utils/data-mock.js

```diff
@@ -949,41 +1005,47 @@ export function getAnalytics4MockPivotResponse( options ) {
 				'RESERVED_MAX',
 				'RESERVED_TOTAL',
 			].reduce( ( acc, aggregate ) => {
-				dimensions.forEach( ( dimension ) => {
-					acc.push( {
-						dimensionValues: [
-							{
-								value: aggregate,
-							},
-							{
-								value: dimension,
-							},
-						],
-						metricValues: generateMetricValues( validMetrics ),
+				options.pivots.forEach( ( { fieldNames } ) => {
+					const pivotDimensionValueSets = fieldNames.map(
+						( dimension ) => allDimensionValues[ dimension ]
+					);
+
+					zip( ...pivotDimensionValueSets ).forEach( ( values ) => {
+						acc.push( {
+							dimensionValues: dimensions.map( ( dimension ) => {
+								const fieldNameIndex =
+									fieldNames.indexOf( dimension );
+
+								return {
+									value:
+										fieldNameIndex === -1
+											? aggregate
+											: values[ fieldNameIndex ],
+								};
+							} ),
+							metricValues: generateMetricValues( validMetrics ),
+						} );
 					} );
 				} );
 				return acc;
 			}, [] );
 
-			// Generate the pivot headers with a row count and pivotDimensionHeaders for each column.
-			data.pivotHeaders = allLimitedDimensionValues.reduce(
-				( acc, dimensionValues ) => {
-					acc.push( {
-						pivotDimensionHeaders: dimensionValues.map(
-							( value ) => ( {
-								dimensionValues: [
-									{
-										value,
-									},
-								],
-							} )
-						),
-						rowCount: dimensionValues.length,
-					} );
-					return acc;
-				},
-				[]
-			);
+			data.pivotHeaders = options.pivots.map( ( { fieldNames } ) => {
```

Comment:
I noticed there was still an issue with `pivotHeaders`.

<details>
<summary>The output was as follows:</summary>

```json 
  "pivotHeaders": [
    {
      "pivotDimensionHeaders": [
        {
          "dimensionValues": [
            {
              "value": "properties/12345/audiences/1"
            }
          ]
        },
        {
          "dimensionValues": [
            {
              "value": "properties/12345/audiences/2"
            }
          ]
        }
      ],
      "rowCount": 2
    },
    {
      "pivotDimensionHeaders": [
        {
          "value": "Dublin"
        },
        {
          "value": "(not set)"
        },
        {
          "value": "Cork"
        }
      ],
      "rowCount": 3
    },
    {
      "pivotDimensionHeaders": [
        {
          "value": "United States"
        },
        {
          "value": "United Kingdom"
        },
        {
          "value": "India"
        }
      ],
      "rowCount": 3
    },
    {
      "pivotDimensionHeaders": [
        {
          "dimensionValues": [
            {
              "value": "Direct"
            }
          ]
        },
        {
          "dimensionValues": [
            {
              "value": "Organic Search"
            }
          ]
        },
        {
          "dimensionValues": [
            {
              "value": "Paid Social"
            }
          ]
        },
        {
          "dimensionValues": [
            {
              "value": "(other)"
            }
          ]
        }
      ],
      "rowCount": 4
    }
  ],
```

</details>

<details>
<summary>But the expectation is for the following shape:</summary>

```json
  "pivotHeaders": [
    {
      "pivotDimensionHeaders": [
        { "dimensionValues": [{ "value": "properties/12345/audiences/1" }] },
        { "dimensionValues": [{ "value": "properties/12345/audiences/2" }] }
      ],
      "rowCount": 2
    },
    {
      "pivotDimensionHeaders": [
        {
          "dimensionValues": [
            { "value": "Dublin" },
            { "value": "United States" }
          ]
        },
        {
          "dimensionValues": [
            { "value": "(not set)" },
            { "value": "United Kingdom" }
          ]
        },
        { "dimensionValues": [{ "value": "Cork" }, { "value": "India" }] }
      ],
      "rowCount": 6
    },
    {
      "pivotDimensionHeaders": [
        { "dimensionValues": [{ "value": "Direct" }] },
        { "dimensionValues": [{ "value": "Organic Search" }] },
        { "dimensionValues": [{ "value": "Paid Social" }] },
        { "dimensionValues": [{ "value": "(other)" }] }
      ],
      "rowCount": 4
    }
  ],
```

</details>

I took a look and have found a fix - in the interest of time I have pushed the change directly (see commit 82463974f19634530512bb6b3636bbdda012e23d). Please review it and if you are happy with the change, update the fixture file accordingly.